### ISSUE TITLE
Date input has appearance and functional issues

### ISSUE THREAD

**Issue by theadamparker** [undp/design-system#1330]
The [input type **date**](https://design.undp.org/?path=/story/components-forms-input-fields--date) has several issues:

- Currently it shows light gray placeholder text, but when a value is input the text remains light gray. The correct color should be black.
- The date input should support a browser's native date selector UI as much as possible. Currently it does not work for Chrome, Firefox has appearance issues (FF adds a second calendar icon), and Safari looks ok.

**Comment by vern-gora** [comment]
@theadamparker, I solved this issue except for 2 nuances.

First: I tried to make the color change work after one of the subfields was filled (day, month, or year), but css does not have the ability to style the subfields, and if you do it through js, it only catches when all the subfields are filled. Now it works so that the gray color changes to black when all the subfields are filled (as shown in the video).


https://github.com/user-attachments/assets/e783367f-5acd-45ce-9f2c-4b8f7bdccb66


Second: I made it so that in all browsers the default calendar icons are hidden behind a custom one, except for firefox. All sources say that it cannot be styled because firefox does not provide the ability to style individual parts of the input[type='date'] with pseudo-elements, so it will always be at the end of the input (as shown in the photo).

<img width="1209" alt="photo" src="https://github.com/user-attachments/assets/1a36919d-225f-4006-85d1-a6f184b434af">
